demisto_sdk/commands/update_xsoar_config_file/update_xsoar_config_file.py

```diff
@@ -0,0 +1,152 @@
+                pack_id = False
+                print_color("Error: Missing option '-pi' / '--pack-id'.", LOG_COLORS.RED)
+            if not self.pack_data:
+                pack_data = False
+                print_color("Error: Missing option '-pd' / '--pack-data'.", LOG_COLORS.RED)
+            if not pack_data or not pack_id:
+                return False
+        return True
+
+    def add_all_installed_packs_to_config_file(self):
+        """
+        Update the MarketPlace Packs Section in the Configuration File with all the installed packs on the machine.
+        """
+        marketplace_packs = self.get_installed_packs()
+        self.generic_update_xsoar_config_data(section_name=MARKETPLACE_PACKS_SECTION, data_to_update=marketplace_packs)
+
+    def get_installed_packs(self) -> List[Dict[str, str]]:
+        """
+        Gets the current installed packs on the machine.
+        """
+        client = demisto_client.configure(verify_ssl=self.insecure)
+        res = client.generic_request('/contentpacks/metadata/installed', "GET")
+        installed_packs = eval(res[0])
+
+        marketplace_packs = [
+            {
+                "id": pack["id"],
+                "version": pack['currentVersion']
+            }
+            for pack in installed_packs
+        ]
+        return marketplace_packs
+
+    def update_marketplace_pack(self):
+        """
+        Add / Update the MarketPlace Packs Section in the Configuration File with the new pack data.
+        """
+        new_pack = {
+            "id": self.pack_id,
+            "version": self.pack_data
+        }
+        self.generic_update_xsoar_config_data(section_name=MARKETPLACE_PACKS_SECTION, data_to_update=new_pack)
+
+    def update_custom_pack(self):
+        """
+        Add / Update the Custom Packs Section in the Configuration File with the new pack data.
+        """
+        new_pack = {
+            "id": self.pack_id,
+            "url": self.pack_data
+        }
+        self.generic_update_xsoar_config_data(section_name=CUSTOM_PACKS_SECTION, data_to_update=new_pack)
+
+    def generic_update_xsoar_config_data(self, section_name, data_to_update):
+        """
+        Add / Update the Configuration File according to the section and data that provided.
+        """
+        config_file_info = self.get_xsoar_config_data()
+        if config_file_info.get(section_name):
+            config_file_info[section_name].append(data_to_update)
+        else:
+            config_file_info[section_name] = [data_to_update] if isinstance(data_to_update, dict) else data_to_update
+        self.set_xsoar_config_data(config_file_info=config_file_info)
+
+    def get_xsoar_config_data(self):
+        config_file_info = {}
+        if os.path.exists(self.file_path):
+            with open(self.file_path, 'r') as config_file:
+                try:
+                    config_file_info = json.load(config_file)
+                except json.JSONDecodeError:
```

Comment:
Why would that happen?

Reply:
In case that the file exist but empty

Reply:
Let's add a comment that says that, or handle it explicitly in the code.
Your call